Packages/com.unity.inputsystem/InputSystem/API/Input.cs

```diff
@@ -0,0 +1,1862 @@
+            s_PointerAction.performed += OnPointerMoved;
+            s_PointerAction.Enable();
+
+            s_ScrollAction = new InputAction("Scroll", InputActionType.PassThrough, expectedControlType: "Vector2");
+            s_ScrollAction.AddBinding("<Mouse>/scroll");
+            s_ScrollAction.performed += OnMouseScrolled;
+            s_ScrollAction.Enable();
+
+            InitializeGlobalActions(defaultGlobalActionsPath, globalActionsAssetPath);
+        }
+
+        internal static void Shutdown()
+        {
+            if (s_PointerAction != null)
+            {
+                s_PointerAction.performed -= OnPointerMoved;
+                s_PointerAction.Disable();
+                s_PointerAction = null;
+            }
+
+            if (s_ScrollAction != null)
+            {
+                s_ScrollAction.performed -= OnMouseScrolled;
+                s_ScrollAction.Disable();
+                s_ScrollAction = null;
+            }
+
+            InputSystem.onDeviceChange -= OnDeviceChange;
+
+            ShutdownGlobalActions();
+        }
+
+        private static void OnPointerMoved(InputAction.CallbackContext context)
+        {
+            s_PointerPosition = context.ReadValue<Vector2>();
+        }
+
+        private static void OnMouseScrolled(InputAction.CallbackContext context)
+        {
+            s_ScrollDelta = context.ReadValue<Vector2>();
+        }
+
+        private static void OnDeviceChange(InputDevice device, InputDeviceChange change)
+        {
+            if (change == InputDeviceChange.Added)
+            {
+                AddGamepadToFirstFreeSlot(device);
+                AddJoystickToFirstFreeSlot(device);
+            }
+            else if (change == InputDeviceChange.Removed || change == InputDeviceChange.Disconnected)
+            {
+                RemoveGamepad(device);
+                RemoveJoystick(device);
+            }
+        }
+
+        private static void AddGamepadToFirstFreeSlot(InputDevice device)
+        {
+            if (!(device is Gamepad gamepad)) return;
+
+            for (var i = 0; i < maxGamepadSlots; i++)
+            {
+                if (s_Gamepads[i] != null) continue;
+
+                var frameCount = Time.frameCount;
+#if UNITY_EDITOR
+                if (!s_TimeHasUpdatedThisFrame)
+                    frameCount += 1;
+#endif
+                s_Gamepads[i] = gamepad;
+                s_GamepadsConnectedFrames[i] = frameCount;
+                break;
+            }
+        }
+
+        private static void RemoveGamepad(InputDevice device)
+        {
+            for (var i = 0; i < maxGamepadSlots; i++)
+            {
+                if (s_Gamepads[i] != device) continue;
+
+                s_Gamepads[i] = null;
+
+                var frameCount = Time.frameCount;
+#if UNITY_EDITOR
+                if (!s_TimeHasUpdatedThisFrame)
+                    frameCount += 1;
+#endif
+                s_GamepadsDisconnectedFrames[i] = frameCount;
```

Comment:
Should we be clearing the s_GamepadsConnectedFrames[i[ value too.
I'm wondering if we would get a bogus connect on frame wrapping the max counter, although that would take along time for an int value so unlikely to occur in practice.

(likewise above clearing s_GamepadsDisconnectedFrames on a connection.